### PR TITLE
Minor documentation additions - Akka Streams RestartFlow and RestartSettings

### PR DESCRIPTION
**Pull request by mhbuck**
## Changes

Addition of method documentation that were found missing when working with restart logic.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [X] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [X] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

### CODE REVIEW

File: src/core/Akka.Streams/Dsl/RestartFlow.cs [hunk 1]
```diff
@@ -162,7 +162,7 @@ public static class RestartFlow
         /// <para>This uses the same exponential backoff algorithm as <see cref="BackoffOptions"/>.</para>
         /// </summary>
         /// <param name="flowFactory">A factory for producing the <see cref="Flow"/>] to wrap.</param>
-        /// <param name="settings"></param>
+        /// <param name="settings"><see cref="RestartSettings" /> defining restart configuration</param>
```
Comment (mhbuck) [inline src/core/Akka.Streams/Dsl/RestartFlow.cs:165]: Matching the documentation for the `WithBackoff` method in this class

File: src/core/Akka.Streams/RestartSettings.cs [hunk 2]
```diff
@@ -26,6 +26,12 @@ private RestartSettings(TimeSpan minBackoff, TimeSpan maxBackoff, double randomF
             MaxRestartsWithin = maxRestartsWithin;
         }
 
+        /// <summary>
+        /// Creates a new instance of <see cref="RestartSettings"/> class 
+        /// </summary>
+        /// <param name="minBackoff">Minimum (initial) duration until the child actor will started again, if it is terminated</param>
+        /// <param name="maxBackoff">The exponential back-off is capped to this duration</param>
+        /// <param name="randomFactor">After calculation of the exponential back-off an additional random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay. In order to skip this additional delay pass in `0`.</param>
```
Comment (mhbuck) [inline src/core/Akka.Streams/RestartSettings.cs:34]: Addition of documentation matching the extension methods in the class

**Review by mhbuck — COMMENTED** [review]
Some minor tweaks that were encountered today doing changes around Restart

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**